clash_lib/src/proxy/utils/test_utils/docker_runner.rs

```diff
@@ -0,0 +1,158 @@
+//! This example will run a non-interactive command inside the container using `docker exec`
+
+use std::collections::HashMap;
+
+use bollard::container::{Config, RemoveContainerOptions};
+use bollard::secret::{HostConfig, Mount, PortBinding};
+use bollard::Docker;
+
+use bollard::exec::{CreateExecOptions, StartExecResults};
+use bollard::image::CreateImageOptions;
+
+use anyhow::Result;
+use futures::{Future, StreamExt, TryStreamExt};
+
+pub struct DockerTestRunner {
+    _instance: Docker,
+    id: String,
+}
+
+impl DockerTestRunner {
+    pub async fn new<
+        'a,
+        T1: serde::ser::Serialize + Into<std::string::String> + std::fmt::Debug + Clone,
```

Comment:
is this `T1` is essentially a `String`? https://docs.rs/bollard/latest/bollard/image/struct.CreateImageOptions.html 

can we simplify the type signature for this function, same to the `Z` below

Reply:
Mostly, this type is String.

The type is written in accordance with the type in bollard, the docker crate.

And I think you are right, we can simplify the signature here.

Reply:
fixed